docs/migration/v12.md

```diff
@@ -0,0 +1,329 @@
+-       userEvent.click(getByRole('button', { name: 'Save comment' }))
+-       expect(getByRole('progressbar')).toBeInTheDocument()
++       // Use the setup from above:
++       const btnSaveComment = await findByRole('button', {
++           name: 'Save comment',
++       })
++       await user.click(btnSaveComment)
++       await findByRole('progressbar')
+    })
+})
+```
+
+#### Enzyme
+
+Enzyme is [considered dead](https://dev.to/wojtekmaj/enzyme-is-dead-now-what-ekl): there is no official adapter for React 18 (or 17), so it's best to [migrate to React Testing Library](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme/) for tests, if possible.
+
+If it's not currently possible to migrate, there is an [unofficial React 18 adapter](https://github.com/cfaester/enzyme-adapter-react-18) that can help in the mean time.
+
+## Environment variables
+
+### `REACT_APP_` prefix no longer needed
+
+Now that we're moving away from Create React App, the formatting for environment variable names has gotten simpler.
+
+As before, environment variables with names that start with `DHIS2_` are added to the app, and can be picked up from `.env` files or on the command line for example. Before, the platform had to add `REACT_APP_` before `DHIS2_` so that Create React App would add the variable to the app, so a variable like `DHIS2_MY_VAR` in the environment would become accessible on `process.env.REACT_APP_DHIS2_MY_VAR` in the app. Now, the `REACT_APP_` prefix isn't needed, so the variable can be accessed on just `process.env.DHIS2_MY_VAR`.
+
+The `REACT_APP_`-prefixed variables are now deprecated, but they will be kept around alongside the shorter `DHIS2_` variables in this version for backwards compatibility. For example, both `process.env.REACT_APP_DHIS2_MY_VAR` and `process.env.DHIS2_MY_VAR` will be available. The `REACT_APP_` prefixed variables will be removed in the next version though, so make sure to migrate to just the shorter `DHIS2_` variable names.
+
+### `import.meta.env`
+
+By default, Vite adds its environment variables to the [`import.meta.env` object](https://vite.dev/guide/env-and-mode#env-variables). The App Platform will add `DHIS2_` env vars to `import.meta.env` as well, and the configured [`envPrefix`](https://vite.dev/config/shared-options.html#envprefix) is `DHIS2_`. Emphasis is given to environment variables on the `process.env` object, however, since it is a more widely-used pattern.
+
+## Troubleshooting & tips
+
+Here are some edge cases that were encountered when upgrading some apps with the new platform version:
+
+-   There is a bug in Vite: importing from a directory, e.g. `‘./app/‘` where we would assume to get `./app/index.[jt]s`, can incorrectly resolve to a file with a similar name to the directory, e.g. `./App.tsx`. I see this as a bug with Vite, but as a workaround, file and dir names can be changed so they don’t conflict. Noticed in the Maintenance app beta.
```

Comment:
Can the Maintenance app example be linked/explained?

Reply:
I think it's probably best just to remove the mention, which I've done -- I think the rest of the comment is enough info. I've also removed similar "noticed in ___" mentions